components/scream/src/share/io/scream_output_manager.cpp

```diff
@@ -287,12 +287,15 @@ void OutputManager::run(const util::TimeStamp& timestamp)
     // We're adding one snapshot to the file
     ++filespecs.num_snapshots_in_file;
 
-    // Now that we've written output to this file we need reset the nsteps.
-    control.nsamples_since_last_write = 0;
-
-    // Since we wrote to file we need to reset the nsamples_since_last_write and the timestamp
+    // Since we wrote to file we need to reset the nsamples_since_last_write, the timestamp
     control.nsamples_since_last_write = 0;
     control.timestamp_of_last_write = timestamp;
+    // and the local views - unless it is a checkpoint step, then we keep local views.
```

Comment:
Seems like the comment sentence is broken up by the two lines of code. I would either put all comments together, or put some `...` at the end of the first and beginning of the latter, to show that it's the same comment.